src/server/pkg/storage/fileset/fileset.go

```diff
@@ -37,6 +37,12 @@ type FileSet struct {
 }
 
 func newFileSet(ctx context.Context, storage *Storage, name string, memThreshold int64, defaultTag string, opts ...Option) *FileSet {
+	// because we don't have the option to return an error, we can't use ctx here.
```

Comment:
This is not the approach we should take. We can change the function signature to return an error, so we can handle the canceled context case.